server/middleware/prepaids.coffee

```diff
@@ -116,3 +124,90 @@ module.exports =
           schoolPrepaidsMap[school].push prepaid
 
     res.send({prepaidActivityMap, schoolPrepaidsMap})
+  
+  # Separate endpoint from legacy prepaid purchase handler
+  purchaseStarterLicenses: wrap (req, res) ->
+    if req.body.type not in ['starter_license']
+      throw new errors.Forbidden("License type invalid: #{req.body.type}")
+
+    creator = req.user
+    maxRedeemers = parseInt(req.body.maxRedeemers)
+    months = parseInt(req.body.months)
+    token = req.body.stripe?.token
+    timestamp = req.body.stripe?.timestamp
+
+    if isNaN(maxRedeemers) or maxRedeemers < 1
+      throw new errors.UnprocessableEntity("Invalid number of licenses to buy: #{maxRedeemers}")
+
+    alreadyOwnedStarterLicenses = yield Prepaid.find({
+      creator: mongoose.Types.ObjectId(creator.id)
+      type: 'starter_license'
+    }).exec()
+    alreadyOwnedStarterLicenseCount = alreadyOwnedStarterLicenses.map((prepaid) -> prepaid.get('maxRedeemers')).reduce(((a,b) -> a+b), 0)
+
+    if maxRedeemers + alreadyOwnedStarterLicenseCount > Prepaid.MAX_STARTER_LICENSES
+      throw new errors.Forbidden('You cannot own more than 75 starter licenses.')
+    
+    if not (token or creator.isAdmin())
+      throw new errors.UnprocessableEntity('Missing required Stripe token')
+
+    if creator.isAdmin()
+      try
+        yield createStarterLicense({ creator: creator.id, maxRedeemers })
+        res.status(200).send(prepaid)
+      catch e
+        throw new errors.InternalServerError("Database error: #{e.message}")
```

Comment:
You don't need to wrap thrown errors in try/catch when within co-express's `wrap` function. It automatically grabs them and sends them through `next` to the next error middleware. That should send 500 (or 409 if Mongo complains of a conflict).